qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java

```diff
@@ -415,6 +415,17 @@ public void test050BasicApiTests() throws Exception {
         runElasticsearchTestsAsElastic(PASSWORD);
     }
 
+    /**
+     * Check that the JDK uses the Cloudflare zlib, instead of the default one.
+     */
+    public void test060JavaUsesCloudflareZlib() throws Exception {
+        waitForElasticsearch(installation, "elastic", PASSWORD);
+
+        final boolean matches = sh.run("bash -c 'pmap -p $(pidof java)'").stdout.lines().anyMatch(line -> line.contains("cloudflare-zlib"));
+
+        assertTrue("Expect java to be using cloudflare-zlib", matches);
```

Comment:
nit: it would be nice to have the output of the command in the error message in case the problem is not fully reproducible?

Reply:
Good idea @jpountz, I've modified the test.